core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -457,15 +457,16 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   private[history] def checkForLogs(): Unit = {
     try {
       val newLastScanTime = clock.getTimeMillis()
+      logDebug("Checking change 2")
       logDebug(s"Scanning $logDir with lastScanTime==$lastScanTime")
 
       val updated = Option(fs.listStatus(new Path(logDir))).map(_.toSeq).getOrElse(Nil)
-        .filter { entry => !isBlacklisted(entry.getPath) }
+        .filter { entry => !entry.isDirectory() && !entry.getPath().getName().startsWith(".") && !isBlacklisted(entry.getPath) }
```

Comment:
Can you add a comment here?

Reply:
oh... thanks for pointing... this is not required... I'll remove it.